pkg/upgrade/container/container_test.go

```diff
@@ -0,0 +1,150 @@
+package container_test
+
+import (
+	"path/filepath"
+
+	upgradeapiv1 "github.com/rancher/system-upgrade-controller/pkg/apis/upgrade.cattle.io/v1"
+	"github.com/rancher/system-upgrade-controller/pkg/upgrade/container"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("Container", func() {
+	var (
+		zeroContainer *corev1.Container
+		testContainer corev1.Container
+		testOption    container.Option
+	)
+	BeforeEach(func() {
+		zeroContainer = &corev1.Container{}
+		testContainer = *zeroContainer
+		Expect(zeroContainer).To(Equal(&testContainer))
+		Expect(testContainer).To(BeZero())
+	})
```

Comment:
Guessing you don't need to teardown ?

Reply:
In the cases of these tests, no, I don't. This is because the `BeforeEach` re-initializes before every test case.